conda/cli/main_config.py

```diff
@@ -731,8 +731,6 @@ def execute_config(args, parser):
     for arg, prepend in zip((args.prepend, args.append), (True, False)):
         for key, item in arg:
             key, subkey = key.split(".", 1) if "." in key else (key, None)
-            if key == "channels" and key not in rc_config:
-                rc_config[key] = ["defaults"]
```

Comment:
Fair enough, but that means we'll have to do something in setup-miniconda to prevent the implicit `defaults` addition. Either the `nodefaults` hack or a boolean to implement this explicit behavior.